Source/Core/InputCommon/ControllerInterface/ControllerInterface.cpp

```diff
@@ -271,6 +273,10 @@ bool ControllerInterface::AddDevice(std::shared_ptr<ciface::Core::Device> device
   if (!m_is_init)
     return false;
 
+  ASSERT_MSG(CONTROLLERINTERFACE, !tls_is_updating_devices,
+             "Devices shouldn't be added within input update calls, there is a risk of deadlock "
+             "if another thread was already here");
```

Comment:
This doesn't seem like an accurate assert message here in `Initialize`?

Reply:
It's in `AddDevice()`

Reply:
Yeah, you're right. I'm struggling to look at github diffs.